### DISCUSSION TITLE
systemd stopped working suddenly

### DISCUSSION THREAD

**Discussion by GordonSmith** [microsoft/WSL#9063] [Q&A] [answered]
Was up and running with a microk8s environment a couple of weeks ago, tried to start it today and am now getting:
```
Interacting with snapd is not yet supported on Windows Subsystem for Linux.
This command has been left available for documentation purposes only.
```

**Comment by GordonSmith** [comment]
The following resolved the issue for me:
```
echo":windowsexe:M::\x4d\x5a::/init:PF"| sudo tee -a /etc/binfmt.d/windowsexe.conf
```

**Comment by nlgl** [comment]
Hello Mr. GordonSmith

Is your environment windoww11+ Ubuntu installed in WSL2  (with the addition to  wsl.conf ref: https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl/  ) ? 

I did what is advised in the ref but I get  "internal error, please report: running "microk8s" failed: cannot find installed snap "microk8s" at revision 4094: missing file /snap/microk8s/4094/meta/snap.yaml " . I tried both 20.4 and 22.4 versions.  

Many thanks in advance.

Ali

**Reply by GordonSmith** [reply]
Yes (Windows 11 insiders beta) + WSL2 (Ubuntu 22.04).  Is this your command line:
```
sudo snap install microk8s --classic --channel=1.25
```
?

**Comment by nlgl** [comment]
Good morning and thank you very much.

Command line is just like below (actually same as yours) :

sudo snap install microk8s --classic --channel=1.25/stable

And I have  /etc/wsl.conf with
[boot]
systemd=true

as written  in  https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl/

And Windows 11 version:

***@***.***


By the way, with https://youtu.be/cWlYe0CE2iU and  https://forum.snapcraft.io/t/running-snaps-on-wsl2-insiders-only-for-now/13033  I managed to install microk8s in to my “old” windows 10 laptop last night.

I am still keen to use windows 11 though for my lab env in my desktop PC with 32 gbyte ram.

Thanks a lot again.

Ali



Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by kampnerj** [comment]
same problem as OP, but the answer did not resolve it for me.

**Reply by kampnerj** [reply]
ok, fixed with the update.

**Comment by NotTheDr01ds** [comment, marked as answer]
@GordonSmith The original problem with Snap should now be fixed, assuming you are running Ubuntu 22.04?  If so, you should be able to:

* `sudo rm /etc/binfmt.d/windowsexe.conf` to remove the temporary workaround
* `sudo apt update && sudo apt upgrade -y` to get Snapd to the latest version with the fix
*  Exit Ubuntu and `wsl --shutdown` (required to reset the `binfmt_misc` change)

After this, `snap` commands should be working again.

**Reply by GordonSmith** [reply]
Tested and working!

**Reply by nlgl** [reply]
Great.

Is your microk8s  cluster  single node or multi node? 

Tnks

**Reply by GordonSmith** [reply]
Single node (inside single WSL instance)
